src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        keys = (K[]) new Object[STORAGE_SIZE];
+        values = (V[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = indexOf(key);
+        if (index != -1) {
+            values[index] = value;
+        } else if (size < STORAGE_SIZE) {
```

The `put` method does not handle the case where the storage is full. If the `size` is equal to `STORAGE_SIZE`, the method should notify the user or handle the situation appropriately, rather than silently failing.